gardenlinux/gardenlinux
# test: refactor test containers

**How to categorize this PR?**

/kind test
/area os
/os garden-linux

**What this PR does / why we need it**:

Refactors the `base-test` and `integration-test` containers:
- clean up the pipfiles
- install azure-cli through pip and not via deb package
- make sure that pip packages pulled in in `base-test` do not get downgraded in `integration-test`

**Which issue(s) this PR fixes**:
Failing integration tests on Azure during the nightly builds.

**Release note**:

```feature user
- test: refactor test containers
```
